worker-document/src/main/java/com/hpe/caf/worker/document/scripting/specs/UrlScriptSpec.java

```diff
@@ -16,37 +16,37 @@
 package com.hpe.caf.worker.document.scripting.specs;
 
 import com.hpe.caf.worker.document.DocumentWorkerScript;
+import com.hpe.caf.worker.document.model.ScriptEngineType;
+
+import java.io.BufferedReader;
 import java.io.IOException;
+import java.io.InputStreamReader;
 import java.io.Reader;
 import java.net.URI;
 import java.net.URISyntaxException;
 import java.net.URL;
+import java.nio.charset.StandardCharsets;
 import java.util.Objects;
 import javax.annotation.Nonnull;
-import javax.script.Compilable;
-import javax.script.CompiledScript;
-import javax.script.ScriptException;
-import jdk.nashorn.api.scripting.URLReader;
 
 public final class UrlScriptSpec extends RemoteScriptSpec
 {
     private final URL url;
     private final URI uri;
+    private final ScriptEngineType engineType;
 
     public UrlScriptSpec(final URL url) throws URISyntaxException
     {
         this.url = Objects.requireNonNull(url);
         this.uri = url.toURI();
+        this.engineType = ScriptEngineType.NASHORN;
     }
 
-    @Nonnull
-    @Override
-    public CompiledScript compile(final Compilable compiler) throws ScriptException
```

Comment:
Shouldn't this implementation still be present for the Nashorn support?

Reply:
Since we are keeping Nashorn around I'd rather that we left the existing implementation alone (because Nashorn's `URLReader` does have some Nashorn-specific enhancements).  So just rename the existing `UrlScriptSpec` class to `NashornUrlScriptSpec` and then introduce a separate new more general `UrlScriptSpec` that uses the `InputStreamReader` like you are doing.